File: sdk/php/src/Pool/ConnectionPoolConfig.php
<?php

namespace HJTPX\Captcha\Pool;

class ConnectionPoolConfig
{
    public $maxConnections;
    public $connectionTimeout;
    public $requestTimeout;

    public function __construct(
        int $maxConnections = 10,
        int $connectionTimeout = 30,
        int $requestTimeout = 30
    ) {
        $this->maxConnections = $maxConnections;
        $this->connectionTimeout = $connectionTimeout;
        $this->requestTimeout = $requestTimeout;
    }
}

File: sdk/csharp/src/Hjtpx.Captcha.Sdk/Pool/ConnectionPoolConfig.cs
namespace Hjtpx.Captcha.Pool;

public class ConnectionPoolConfig
{
    public int MaxConnections { get; set; } = 100;
    public int MaxConnectionsPerRoute { get; set; } = 50;
    public int ConnectionTimeoutMs { get; set; } = 5000;
    public int SocketTimeoutMs { get; set; } = 30000;
    public int ConnectionRequestTimeoutMs { get; set; } = 5000;
    public int TimeToLiveMs { get; set; } = 60000;
    public bool ValidateAfterInactivity { get; set; } = true;
    public int ValidateAfterInactivityMs { get; set; } = 2000;
}
